src/espresso-migration/Migration.sol

```diff
@@ -0,0 +1,39 @@
+pragma solidity ^0.8.9;
+
+import "../challenge/IChallengeManager.sol";
+import "../rollup/IRollupAdmin.sol";
+
+import "@openzeppelin/contracts/proxy/transparent/ProxyAdmin.sol";
+import "@openzeppelin/contracts/utils/Address.sol";
+
+contract Migration{ 
+    function perform(
+        IRollupCore rollup,
+        address proxyAdmin,
+        bytes32 newWasmModuleRoot,
+        bytes32 currentWasmModuleRoot,
+        address newOspEntry,
+        address currentOspEntry
+    ) public{
+        //Handle assertions in the perform functoin as we shouldn't be storing local state for delegated calls.
+        require(newWasmModuleRoot != bytes32(0), "_newWasmModuleRoot cannot be empty");
```

Comment:
nit: I think we should use custom errors to save gas as we have quite a lot of `require`'s here

Reply:
I agree that this would be rather helpful to reduce the gas used.

Reply:
I think for these contracts the gas usage is not of much concern because each chain only does the upgrade once, and we can just go for whatever is simplest to write.